lib/tenstorrent/bm_event/Kconfig

```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2024 Tenstorrent AI ULC
+# SPDX-License-Identifier: Apache-2.0
+
+config TT_BM_EVENT
+	bool "BM event support"
+	help
+	  BM event support.
+
+module = TT_BM_EVENT
+module-str = TT BM event
+source "subsys/logging/Kconfig.template.log_config"
```

Comment:
This is a really good idea - you can declare the bm_event log source with this in a header

```cpp
#include <zephyr/logging/log.h>

LOG_MODULE_DECLARE(bm_event);
```

And then in a c file (approximately) something like
```cpp
#include <zephyr/logging/log.h>

LOG_MODULE_REGISTER(bm_event, CONFIG_BM_EVENT_LOG_LEVEL);
```

and then you can `LOG_INFO(..)` anywhere. I don't know if there is any possibility of error, but it's always a good idea to log errors (at least until they are very rare, and then we can change them later to `LOG_DBG(...)`